doc/dev/release.md

```diff
@@ -0,0 +1,259 @@
+# Releases
+
+Making an Operator SDK release involves:
+
+- Updating `CHANGELOG.md`.
+- Tagging and signing a git commit and pushing the tag to GitHub.
+- Building a release binary and signing the binary
+- Creating a release by uploading binary, signature, and `CHANGELOG.md` updates for the release to GitHub.
+- Creating a patch version branch of the form `v1.2.x` for major and minor releases.
+
+Releases can only be performed by [maintainers][doc-maintainers].
+
+## Dependency and platform support
+
+### Go version
+
+Release binaries will be built with the Go compiler version specified in the Operator SDK's [prerequisites section][doc-readme-prereqs].
+
+### Kubernetes versions
+
+As the Operator SDK interacts directly with the Kubernetes API, certain API features are assumed to exist in the target cluster. The currently supported Kubernetes version will always be listed in the SDK [prerequisites section][doc-readme-prereqs].
+
+### Operating systems and architectures
+
+Release binaries will be built for the `x86_64` architecture for both GNU Linux and MacOS Darwin platforms.
+
+Support for the Windows platform or any architecture other than `x86_64` is not on the roadmap at this time.
+
+## Binaries and signatures
+
+Binaries will be signed using a maintainers' verified GitHub PGP key. Both binary and signature will be uploaded to the release. Ensure you import maintainer keys to verify release binaries.
+
+## Release tags
+
+Every release will have a corresponding git semantic version tag beginning with `v`, ex. `v1.2.3`.
+
+Make sure you've [uploaded your GPG key][link-github-gpg-key-upload] and configured git to [use that signing key][link-git-config-gpg-key] either globally or for the Operator SDK repository. Tagging will be handled by `release.sh`.
+
+**Note:** the email the key is issued for must be the email you use for git.
+
+```bash
+$ git config [--global] user.signingkey "$GPG_KEY_ID"
+$ git config [--global] user.email "$GPG_EMAIL"
+```
+
+## Release Notes
+
+Release notes should be a thorough description of changes made to code, documentation, and design. Individual changes, such as bug fixes, should be given their own bullet point with a short description of what was changed. Issue links and handle of the developer who worked on the change should be included whenever possible.
```

Comment:
I thought the release notes would just be the CHANGELOG maintained since the last release.
And we've been following the CHANGELOG format largely based off https://keepachangelog.com/en/1.0.0/#how

I'm not opposed to the format below but I'm not sure we'll be able to maintain it. Or generate it easily enough from the CHANGELOG. Plus is it worth having two formats for the CHANGELOG and the release notes?

It's been my intention that we keep the CHANGELOG simple enough so we keep it updated as much as possible and then update and copy it over at the time of the release.

Reply:
This description is an artifact from the previous release doc. I'll make an update. Nice catch.